### PR TITLE
vault 1.4.0

### PR DESCRIPTION
**Pull request by ianferguson**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
Does this build with a Go version that is still actively supported? (since go 1.12 isn't)

**Comment by SMillerDev** [comment]
```
==> Checking that build is using go version >= 1.13.7...
==> Using go version 1.13.9...
verifying github.com/hashicorp/vault-plugin-database-mongodbatlas@v0.1.0/go.mod: checksum mismatch
	downloaded: h1:HTXNzFr/SAVtJOs7jz0XxZ69jlKtaceEwp37l86UAQ0=
	go.sum:     h1:MP3kfr0N+7miOTZFwKv952b9VkXM4S2Q6YtQCiNKWq8=

SECURITY ERROR
This download does NOT match an earlier download recorded in go.sum.
The bits may have been replaced on the origin server, or an attacker may
have intercepted the download attempt.

For more information, see 'go help module-auth'.
make: *** [prep] Error 1
```

**Comment by ianferguson** [comment]
Sorry for not responding directly yesterday, @SMillerDev.

I bumped the build version as you saw and got the error you put in the comment. I've seen that before going from go modules usage in golang v1.12 to v1.13+, off hand I thought Hashicorp Vault was being built with Golang v1.13 now, but maybe I was wrong -- I should be able to spend some time tracking this down tomorrow or this weekend if no one else beats me to it.

**Comment by as0bu** [comment]
@SMillerDev and @ianferguson that version of Vault does require Go 1.13.7+. I'm not sure why the build is picking up v0.1.0 for that plugin while the current go.sum file in the Vault repo has v0.1.1 for that.

**Comment by SMillerDev** [comment]
Can you implement the `GOPRIVATE` variable like macports did. With a comment that it should be removed for 1.4.1?

File: Formula/vault.rb [hunk 1]
```diff
@@ -16,12 +16,24 @@ class Vault < Formula
...
   end
 
-  depends_on "go@1.12" => :build
+  depends_on "go@1.13" => :build
   depends_on "gox" => :build
 
   def install
     ENV["GOPATH"] = buildpath
 
+    # GOPRIVATE should be removed when v1.4.1 is released.
```
Comment (SMillerDev) [inline Formula/vault.rb:25]: Can you reduce the comment to just this and the macports URL?

**Review by SMillerDev — COMMENTED** [review]
Almost ready

File: Formula/vault.rb (hunk 1 above)
Comment (ianferguson) [reply Formula/vault.rb:25]: done

**Comment by SMillerDev** [comment]
One last thing before merging, could you squash your commits into one `<name> <version>` commit?

**Comment by ianferguson** [comment]
@SMillerDev done

**Comment by SMillerDev** [comment]
Thanks @ianferguson ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Comment by ianferguson** [comment]
@SMillerDev thank you for your review and help, happy to contribute. Also a big thanks to @as0bu for directly linking in the solution and relevant upstream issues!

**Pull request closed without merging**